zypp/PoolQuery.cc

```diff
@@ -316,6 +313,56 @@ namespace zypp
       Predicate        predicate;
       std::string      predicateStr;
       ResKind          kindPredicate = ResKind::nokind;	// holds the 'kind' part if SolvAttr:name looks for an explicit 'kind:name'
+
+    private:
+      /** Serialize \ref strMatcher \ref Match::Mode */
+      static std::string serializeMode( Match::Mode mode_r )
+      {
+	// Legacy code used "[C|X]" to differ just between OTHER (need to (C)ompile) and
+	// using the default search mode. As we now allow to specify a SEARCHMODE we
+	// need to serialize it:
+	switch ( mode_r )
+	{
+#define OUTS(M,S) case Match::M: return #S; break
```

Comment:
I don't think the #define here adds a lot in terms of readability. IMHO defines should be avoided as much as possible, they can be hard to read and especially they are hard to debug.